snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedTransientSnapshot.java

```diff
@@ -75,6 +77,15 @@ private void takeInternal(final Predicate<Path> takeSnapshot) {
     }
   }
 
+  private void calculateAndPersistChecksum() throws IOException {
+    if (!directory.toFile().exists() || directory.toFile().listFiles().length == 0) {
+      // Do not calculate checksum if directory is empty
```

Comment:
Shouldn't the snapshot be invalid if there are no files?

Reply:
I thought about it. But was not sure what should be the semantics. I will change the semantics to mark the snapshot invalid if there are no files.